### ISSUE TITLE
prevent cycle of empty iterable

### ISSUE THREAD

**Issue by xgbuils** [benji6/imlazy#14]
Hi,

Implementation of cycle:
``` javascript
module.exports.cycle = xs => genToIter(function * () {
  while (true) yield * xs
})
```

produces infinite loop that could be avoided:

``` javascript
[...cycle([])] // could be [] but produces and infinite loop.
```

**Comment by benji6** [comment]
Hey awesome catch, I didn't think of this! I've added a test case and a fix here #15 I think it makes sense to take the Clojure approach and return an empty iterable if `cycle` is called with an empty iterable

**Comment by xgbuils** [comment]
Thanks @benji6, nice solution!
